### PR TITLE
Server scaffolding refinement

### PR DESCRIPTION
**Pull request by GGP1**
|Related issue|
|---|
| Closes https://github.com/wazuh/wazuh/issues/27285 |

## Description

Changes the groups files folder name from `shared` to `groups` and moves all certificates to the `/etc/wazuh-server/certs` folder.

## Tests

<details><summary>/etc/wazuh-server folder structure</summary>

```console
root@wazuh-manager:/# ls -la /etc/wazuh-server/
total 44
drwxr-xr-x 1 root  root  4096 Dec 13 16:21 .
drwxr-xr-x 1 root  root  4096 Dec 13 16:21 ..
drwxr-x--- 3   999   999 4096 Dec 13 16:20 api
drwxr-x--- 1   999   999 4096 Dec 13 16:21 certs
drwxr-x--- 2   999   999 4096 Dec 13 16:20 cluster
drwxr-x--- 2   999   999 4096 Dec 13 16:20 groups
-rw-r----- 1 wazuh wazuh  237 Dec 13 16:21 private_key.pem
-rw-r----- 1 wazuh wazuh  174 Dec 13 16:21 public_key.pem
-rw-r----- 1   999   999  703 Dec 13 16:21 wazuh-server.yml
```

</details>

<details><summary>Certificates folder</summary>

```console
root@wazuh-manager:/# ls -la /etc/wazuh-server/certs
total 40
drwxr-x--- 1    999    999 4096 Dec 13 16:21 .
drwxr-xr-x 1 root   root   4096 Dec 13 16:21 ..
-r-------- 1 wazuh  wazuh  1233 Dec 13 16:21 api.crt
-r-------- 1 wazuh  wazuh  1704 Dec 13 16:21 api.key
-rw------- 1 ubuntu ubuntu 1679 Dec 12 18:09 indexer-key.pem
-rw-r--r-- 1 ubuntu ubuntu 1415 Dec 12 18:09 indexer.pem
-rw-r--r-- 1 ubuntu ubuntu 1338 Dec 12 18:09 root-ca.pem
-rw------- 1 ubuntu ubuntu 1708 Dec 12 18:09 server.key
-rw-r--r-- 1 ubuntu ubuntu 1415 Dec 12 18:09 server.pem
```

</details>

<details><summary>Group files folder</summary>

```console
root@wazuh-manager:/# ls -la /etc/wazuh-server/groups
total 12
drwxr-x--- 2  999  999 4096 Dec 13 16:20 .
drwxr-xr-x 1 root root 4096 Dec 13 16:21 ..
```

</details>

<details><summary>Create group</summary>

`POST /groups`

```json
Body
{
  "group_id": "group1"
}

Response
{
    "message": "Group 'group1' created.",
    "error": 0
}
```

</details>

<details><summary>Group files folder</summary>

```console
root@wazuh-manager:/# ls -la /etc/wazuh-server/groups
total 16
drwxr-x--- 1   999   999 4096 Dec 13 16:25 .
drwxr-xr-x 1 root  root  4096 Dec 13 16:21 ..
-rw-rw---- 1 wazuh wazuh   21 Dec 13 16:25 group1.yml
root@wazuh-manager:/# cat /etc/wazuh-server/groups/group1.yml 
# Group configuration
```

```console
root@wazuh-worker1:/# ls -la /etc/wazuh-server/groups/
total 16
drwxr-x--- 1   999   999 4096 Dec 13 16:26 .
drwxr-xr-x 1 root  root  4096 Dec 13 16:21 ..
-rw-rw---- 1 wazuh wazuh   21 Dec 13 16:26 group1.yml
root@wazuh-worker1:/# cat /etc/wazuh-server/groups/group1.yml 
# Group configuration
```

</details>

<details><summary>Get groups</summary>

`GET /groups`

```json
{
    "data": {
        "affected_items": [
            {
                "name": "group1",
                "configSum": "65e363d76ef3e3b76f87b8eaa24cbff9"
            }
        ],
        "total_affected_items": 1,
        "total_failed_items": 0,
        "failed_items": []
    },
    "message": "All selected groups information was returned",
    "error": 0
}
```

</details>

<details><summary>Groups files synchronization</summary>

```console
2024/12/13 16:26:04 INFO: [Worker worker1] [Integrity sync] Starting.
2024/12/13 16:26:04 INFO: [Worker worker1] [Integrity sync] Files to create in worker: 1 | Files to update in worker: 0 | Files to delete in worker: 0
2024/12/13 16:26:04 DEBUG: [Worker worker1] [Integrity sync] Compressing files and 'files_metadata.json' of 1 files.
2024/12/13 16:26:04 DEBUG: [Worker worker1] [Integrity sync] Sending zip file.
2024/12/13 16:26:04 DEBUG: [Worker worker1] [Integrity sync] Zip file sent.
2024/12/13 16:26:04 INFO: [Worker worker1] [Integrity sync] Finished in 0.005s
```

```console
2024/12/13 16:26:04 INFO: [Worker worker1] [Integrity sync] Starting.
2024/12/13 16:26:04 INFO: [Worker worker1] [Integrity sync] Files to create: 1 | Files to update: 0 | Files to delete: 0
2024/12/13 16:26:04 DEBUG: [Worker worker1] [Integrity sync] Worker does not meet integrity checks. Actions required.
2024/12/13 16:26:04 DEBUG: [Worker worker1] [Integrity sync] Updating local files: Start.
2024/12/13 16:26:04 TRACE: [Worker worker1] [Integrity sync] {'groups/group1.yml': ['Processing file groups/group1.yml']}
2024/12/13 16:26:04 DEBUG: [Worker worker1] [Integrity sync] Updating local files: End.
2024/12/13 16:26:04 INFO: [Worker worker1] [Integrity sync] Finished in 0.003s.
```

</details>


<details><summary>Unit tests</summary>

```console
(venv) gasti@gasti:~/work/wazuh$ pytest framework/wazuh/core/tests/test_agent.py::test_get_groups --disable-warnings
============================================================================================================ test session starts =============================================================================================================
platform linux -- Python 3.10.14, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/gasti/work/wazuh/framework
configfile: pytest.ini
plugins: asyncio-0.18.1, tavern-1.23.5, html-2.1.1, metadata-3.1.1, cov-4.1.0, anyio-4.1.0, trio-0.8.0, aiohttp-1.0.4
asyncio: mode=auto
collected 1 item                                                                                                                                                                                                                             

framework/wazuh/core/tests/test_agent.py .                                                                                                                                                                                             [100%]

======================================================================================================= 1 passed, 7 warnings in 0.49s ========================================================================================================
(venv) gasti@gasti:~/work/wazuh$ pytest framework/wazuh/core/tests/test_agent.py::test_agent_delete_single_group --disable-warnings
============================================================================================================ test session starts =============================================================================================================
platform linux -- Python 3.10.14, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/gasti/work/wazuh/framework
configfile: pytest.ini
plugins: asyncio-0.18.1, tavern-1.23.5, html-2.1.1, metadata-3.1.1, cov-4.1.0, anyio-4.1.0, trio-0.8.0, aiohttp-1.0.4
asyncio: mode=auto
collected 1 item                                                                                                                                                                                                                             

framework/wazuh/core/tests/test_agent.py .                                                                                                                                                                                             [100%]

======================================================================================================= 1 passed, 7 warnings in 0.46s ========================================================================================================

(venv) gasti@gasti:~/work/wazuh$ pytest framework/wazuh/core/tests/test_configuration.py::test_get_group_conf --disable-warnings
============================================================================================================ test session starts =============================================================================================================
platform linux -- Python 3.10.14, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/gasti/work/wazuh/framework
configfile: pytest.ini
plugins: asyncio-0.18.1, tavern-1.23.5, html-2.1.1, metadata-3.1.1, cov-4.1.0, anyio-4.1.0, trio-0.8.0, aiohttp-1.0.4
asyncio: mode=auto
collected 1 item                                                                                                                                                                                                                             

framework/wazuh/core/tests/test_configuration.py .                                                                                                                                                                                     [100%]

============================================================================================================= 1 passed in 0.27s ==============================================================================================================
(venv) gasti@gasti:~/work/wazuh$ pytest framework/wazuh/core/tests/test_configuration.py::test_update_group_configuration --disable-warnings
============================================================================================================ test session starts =============================================================================================================
platform linux -- Python 3.10.14, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/gasti/work/wazuh/framework
configfile: pytest.ini
plugins: asyncio-0.18.1, tavern-1.23.5, html-2.1.1, metadata-3.1.1, cov-4.1.0, anyio-4.1.0, trio-0.8.0, aiohttp-1.0.4
asyncio: mode=auto
collected 1 item                                                                                                                                                                                                                             

framework/wazuh/core/tests/test_configuration.py .                                                                                                                                                                                     [100%]

============================================================================================================= 1 passed in 0.26s ==============================================================================================================
(venv) gasti@gasti:~/work/wazuh$ pytest framework/wazuh/core/tests/test_configuration.py::test_update_group_file --disable-warnings
============================================================================================================ test session starts =============================================================================================================
platform linux -- Python 3.10.14, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/gasti/work/wazuh/framework
configfile: pytest.ini
plugins: asyncio-0.18.1, tavern-1.23.5, html-2.1.1, metadata-3.1.1, cov-4.1.0, anyio-4.1.0, trio-0.8.0, aiohttp-1.0.4
asyncio: mode=auto
collected 1 item                                                                                                                                                                                                                             

framework/wazuh/core/tests/test_configuration.py .                                                                                                                                                                                     [100%]

============================================================================================================= 1 passed in 0.27s ==============================================================================================================
(venv) gasti@gasti:~/work/wazuh$ pytest framework/wazuh/core/tests/test_utils.py::test_get_group_file_path --disable-warnings
============================================================================================================ test session starts =============================================================================================================
platform linux -- Python 3.10.14, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/gasti/work/wazuh/framework
configfile: pytest.ini
plugins: asyncio-0.18.1, tavern-1.23.5, html-2.1.1, metadata-3.1.1, cov-4.1.0, anyio-4.1.0, trio-0.8.0, aiohttp-1.0.4
asyncio: mode=auto
collected 1 item                                                                                                                                                                                                                             

framework/wazuh/core/tests/test_utils.py .                                                                                                                                                                                             [100%]

============================================================================================================= 1 passed in 0.36s ==============================================================================================================
```

> Couldn't execute other tests because of https://github.com/wazuh/wazuh/issues/26725.

</details>

### CODE REVIEW

File: api/Makefile [hunk 1]
```diff
@@ -29,7 +29,6 @@ install:
 	# ETC
 	$(INSTALL_DIR) $(ETC_INSTALLDIR)/api
 	$(INSTALL_RW_DIR) $(ETC_INSTALLDIR)/api/configuration
-	$(INSTALL_RW_DIR) $(ETC_INSTALLDIR)/api/configuration/ssl
 	$(INSTALL_RW_DIR) $(ETC_INSTALLDIR)/api/configuration/security
```
Comment (nico-stefani) [inline api/Makefile:32]: Is there any reason to maintain the `api` directory? After moving the certs, only empty directories remain, we could remove these now.

File: api/api/constants.py [hunk 2]
```diff
@@ -3,17 +3,13 @@
 # This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
 from wazuh.core import common
 
-API = 'api'
-API_PATH = common.WAZUH_SHARE / API
-CONFIG_PATH = common.WAZUH_ETC / API / 'configuration'
-CONFIG_FILE_PATH = CONFIG_PATH / 'api.yaml'
-SECURITY_PATH = CONFIG_PATH / 'security'
+CERTS_PATH = common.WAZUH_ETC / 'certs'
+SECURITY_PATH = common.WAZUH_ETC / 'api' / 'configuration' / 'security'
```
Comment (nico-stefani) [inline api/api/constants.py:7]: Related to the previous comment, this should be removed too.

**Review by nico-stefani — CHANGES_REQUESTED** [review]

File: api/Makefile (hunk 1 above)
Comment (GGP1) [reply api/Makefile:32]: The RBAC security.yaml configuration file is still there

File: framework/Makefile [hunk 3]
```diff
@@ -52,7 +52,8 @@ install:
 	# ETC
 	$(INSTALL_DIR) $(ETC_INSTALLDIR)
 	$(INSTALL_DIR) $(ETC_INSTALLDIR)/cluster
-	$(INSTALL_DIR) $(ETC_INSTALLDIR)/shared
+	$(INSTALL_DIR) $(ETC_INSTALLDIR)/certs
+	$(INSTALL_DIR) $(ETC_INSTALLDIR)/groups
```
Comment (nico-stefani) [inline framework/Makefile:56]: Also is needed to modify the [SPEC file](https://github.com/wazuh/wazuh/blob/5245e81db6175631b60d9432d0849a2eac437f4d/packages/rpms/SPECS/wazuh-server.spec#L243) for RPM packages

**Review by nico-stefani — CHANGES_REQUESTED** [review]

File: api/Makefile (hunk 1 above)
Comment (GGP1) [reply api/Makefile:32]: I thought we had a `security.yaml` file there but that configuration is now in `wazuh-server.yml` so I have deleted it from the Makefile 👍🏼

**Review by nico-stefani — APPROVED** [review]
LGTM

**Review by fdalmaup — APPROVED** [review]
LGTM!

**Pull request merged**